ext/fs/std_fs.rs

```diff
@@ -1047,15 +1047,19 @@ fn open_options(options: OpenOptions) -> fs::OpenOptions {
 }
 
 #[inline(always)]
-fn open_with_access_check(
+pub fn open_with_access_check(
   options: OpenOptions,
   path: &Path,
   access_check: Option<AccessCheckCb>,
 ) -> FsResult<std::fs::File> {
   if let Some(access_check) = access_check {
     let path_bytes = path.as_os_str().as_encoded_bytes();
     let is_windows_device_path = cfg!(windows)
-      && path_bytes.starts_with(br"\\.\")
+      && (path_bytes.starts_with(br"\\.\")
+        || path_bytes.starts_with(b"//./")
+        || path_bytes.starts_with(b"//?/")
+        || path_bytes.starts_with(br"\\?\")
+        || path_bytes.ends_with(b"$"))
```

Comment:
`ends_with(b"$")` seems to broad.

To my knowledge only `CONIN$` and `CONOUT$` are the only `$`-ending special file names. Otherwise, files ending with `$` are allowed and are regular files.